src/Policies.cpp

```diff
@@ -117,277 +192,285 @@ std::vector<Policy *> PolicyManager::load(const YAML::Node &policy_yaml)
             }
+            input_ptr = input_stream.get();
+        } catch (std::runtime_error &e) {
+            throw PolicyException(fmt::format("unable to instantiate tap '{}': {}", tap_name, e.what()));
         }
+    }
+
+    InputEventProxy *input_event_proxy = nullptr;
+    try {
+        input_event_proxy = input_ptr->add_event_proxy(tap_filter);
+    } catch (ConfigException &e) {
+        throw PolicyException(fmt::format("unable to create event proxy due to invalid input filter config: {}", e.what()));
+    }
+
+    // if and only if policy succeeds, we will return this in result set
+    policy_ptr->set_tap(tap);
+    policy_ptr->set_input_stream(input_ptr);
+
+    // Handler Section
+
+    auto handler_node = policy_yaml["handlers"];
+    if (!handler_node || !handler_node.IsMap()) {
+        throw PolicyException("missing or invalid handler configuration at key 'handlers'");
+    }
+
+    bool handler_sequence = false;
+    if (!handler_node["modules"] || (!handler_node["modules"].IsMap() && !handler_node["modules"].IsSequence())) {
+        throw PolicyException("missing or invalid handler modules at key 'modules'");
+    } else if (handler_node["modules"].IsSequence()) {
+        handler_sequence = true;
+    }
+
+    policy_ptr->set_modules_sequence(handler_sequence);
 
-        InputEventProxy *input_event_proxy = nullptr;
+    Config window_config;
+    if (handler_node["window_config"] && handler_node["window_config"].IsMap()) {
         try {
-            input_event_proxy = input_ptr->add_event_proxy(tap_filter);
+            window_config.config_set_yaml(handler_node["window_config"]);
         } catch (ConfigException &e) {
-            throw PolicyException(fmt::format("unable to create event proxy due to invalid input filter config: {}", e.what()));
+            throw PolicyException(fmt::format("invalid stream handler window config: {}", e.what()));
         }
-
-        // Handler type
-        if (!it->second["handlers"] || !it->second["handlers"].IsMap()) {
-            throw PolicyException("missing or invalid handler configuration at key 'handlers'");
-        }
-        auto handler_node = it->second["handlers"];
-        bool handler_sequence = false;
-        if (!handler_node["modules"] || (!handler_node["modules"].IsMap() && !handler_node["modules"].IsSequence())) {
-            throw PolicyException("missing or invalid handler modules at key 'modules'");
-        } else if (handler_node["modules"].IsSequence()) {
-            handler_sequence = true;
+    } else {
+        window_config.config_set<uint64_t>("num_periods", _default_num_periods);
+        window_config.config_set<uint64_t>("deep_sample_rate", _default_deep_sample_rate);
+    }
+    window_config.config_set<std::string>("_internal_tap_name", tap_name);
```

Comment:
@weyrick I've added this internal config to pass the tap name to abstractManager so I did not need to add new `tap_name` parameter to Handlers constructor. What do you think?